cmd/runtimecfg/node-ip.go

```diff
@@ -91,53 +94,79 @@ func set(cmd *cobra.Command, args []string) error {
 	}
 	log.Infof("Chosen Node IPs: %v", chosenAddresses)
 
-	// Kubelet
-	kubeletOverrideDir := filepath.Dir(kubeletSvcOverridePath)
-	err = os.MkdirAll(kubeletOverrideDir, 0755)
-	if err != nil {
-		return err
-	}
-	log.Infof("Opening Kubelet service override path %s", kubeletSvcOverridePath)
-	kOverride, err := os.Create(kubeletSvcOverridePath)
-	if err != nil {
-		return err
-	}
-	defer kOverride.Close()
-
 	nodeIP := chosenAddresses[0].String()
 	nodeIPs := nodeIP
 	if len(chosenAddresses) > 1 {
 		nodeIPs += "," + chosenAddresses[1].String()
 	}
+	// Kubelet
 	kOverrideContent := fmt.Sprintf("[Service]\nEnvironment=\"KUBELET_NODE_IP=%s\" \"KUBELET_NODE_IPS=%s\"\n", nodeIP, nodeIPs)
 	log.Infof("Writing Kubelet service override with content %s", kOverrideContent)
-	_, err = kOverride.WriteString(kOverrideContent)
+	err = writeToFile(kubeletSvcOverridePath, kOverrideContent)
 	if err != nil {
 		return err
 	}
 
 	// CRI-O
-	crioOverrideDir := filepath.Dir(crioSvcOverridePath)
-	err = os.MkdirAll(crioOverrideDir, 0755)
+	cOverrideContent := fmt.Sprintf("[Service]\nEnvironment=\"CONTAINER_STREAM_ADDRESS=%s\"\n", nodeIP)
+	log.Infof("Writing CRIO service override with content %s", cOverrideContent)
+	err = writeToFile(crioSvcOverridePath, cOverrideContent)
 	if err != nil {
 		return err
 	}
-	log.Infof("Opening CRI-O service override path %s", crioSvcOverridePath)
-	cOverride, err := os.Create(crioSvcOverridePath)
+
+	// node ip hint for all other services
+	err = writeToFile(nodeIpFile, nodeIP)
 	if err != nil {
 		return err
 	}
-	defer cOverride.Close()
 
-	cOverrideContent := fmt.Sprintf("[Service]\nEnvironment=\"CONTAINER_STREAM_ADDRESS=%s\"\n", chosenAddresses[0])
-	log.Infof("Writing CRI-O service override with content %s", cOverrideContent)
-	_, err = cOverride.WriteString(cOverrideContent)
+	for i := 0; i < len(chosenAddresses) && i < 2; i++ {
+		if utils.IsIPv6(chosenAddresses[i]) && !fileExists(nodeIpIpV6File) {
```

Comment:
Is there a race condition between the `os.Stat` and `os.Create`?  I don't think it is atomic.   

We could maybe just check for `EEXIST` from the `os.Create` error using `errors.Is(err, fs.ErrExist)`

Reply:
i will remove file check, will just set boolean will be more obvious i assume